### PR TITLE
Priority hightlight

### PR DESCRIPTION
**Pull request by OliverGreen27**
clicking on mutant in file report with mouse now shows the smallest by text length mutant. List of mutants in info-box is also sorted by smallest text length

### CODE REVIEW

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 1]
```diff
@@ -566,21 +566,29 @@ struct MetaSpan {
 
     StatusColor status;
     string onClick;
+    static struct T {
+        ulong length = -1;
+        MutationId id;
+        ubyte status = Mutation.Status.killed;
```
[the remark this answers is not in the document]
Comment (OliverGreen27) [reply plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:572]: is it possible to reference the specific enum type from the type.d file then? or do I use enum StatusColor that also recides in MetaSpan and just reorder the values(alive,kiled, etc...) cause they are in the wrong order

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 2]
```diff
@@ -564,23 +564,38 @@ struct MetaSpan {
...
+        if (muts.length != 0) {
+            clickPrio.length = muts[0].txt.mutation.length;
+            clickPrio.id = muts[0].id;
+            clickPrio.status = muts[0].mut.status;
+        }
         foreach (ref const m; muts) {
             status = pickColor(m, status);
-            if (onClick.length == 0 && m.mut.status == Mutation.Status.alive) {
-                onClick = format(click_fmt2, m.id.get);
+            if (m.mut.status == clickPrio.status
```
Comment (joakim-brannstrom) [inline plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:588]: Need to impl the previous algo:


1. priorities alive mutants over everything else
2. priorities the shortest alive
3. priorities shortest mutant (which mean that if there are no alive mutants on the line then it is only the size)

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 3]
```diff
@@ -564,23 +564,37 @@ struct MetaSpan {
         noCoverage,
     }
 
+    static struct MutationLength {
+        ulong length;
+        MutationId id;
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:569]: should probably be MutationStatusId considering Viktors changes?

File: plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d [hunk 4]
```diff
@@ -564,23 +564,37 @@ struct MetaSpan {
...
+        if (muts.length != 0) {
+            clickPrio = MutationLength(muts[0].txt.mutation.length, muts[0].id, muts[0].mut.status);
+        }
         foreach (ref const m; muts) {
             status = pickColor(m, status);
-            if (onClick.length == 0 && m.mut.status == Mutation.Status.alive) {
-                onClick = format(click_fmt2, m.id.get);
+            if (m.mut.status == Mutation.Status.alive && clickPrio.status != Mutation.Status.alive) {
+                clickPrio = MutationLength(m.txt.mutation.length, m.id, m.mut.status);
+            } else if (m.txt.mutation.length < clickPrio.length && m.mut.status == clickPrio.status) {
```
Comment (joakim-brannstrom-work1) [inline plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:593]: questions.
1. the desired behavior is that the smallest alive is shown. i think the impl handle that.
2. if there are no alive then the desired is that the smallest mutant is shown irregardless of status.
- i do not think the impl that that case?

Comment (OliverGreen27) [reply plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d:593]: ok so just removing the last statement in the elseif statement would fix that right?

Comment (OliverGreen27) [reply plugin/mutate/source/dextool/plugin/mutate/backend/report/html/package.d]: the algo was wrong, testing helped to build it better. Thank you!

**Pull request merged**